R/deps.R

```diff
@@ -280,25 +280,39 @@ update.package_deps <- function(object, ..., quiet = FALSE, upgrade = TRUE) {
 
 install_packages <- function(packages, repos = getOption("repos"),
                              type = getOption("pkgType"), ...,
-                             dependencies = FALSE, quiet = NULL,
-                             # These are options to `install()` used when
-                             # installing remotes, but can get passed to us by
-                             # `...` so we just ignore them here
-                             build, build_opts) {
+                             dependencies = FALSE, quiet = NULL) {
+
+  # We want to pass only args that exist in the downstream functions
+  args_to_keep <-
+    unique(
+      names(
+        c(
+          formals(install.packages),
+          formals(download.file)
+        )
+      )
+    )
+
+  args <- list(...)
+  args <- args[names(args) %in% args_to_keep]
 
   if (is.null(quiet))
     quiet <- !identical(type, "source")
 
   message("Installing ", length(packages), " packages: ",
     paste(packages, collapse = ", "))
 
-  safe_install_packages(
-    packages,
-    repos = repos,
-    type = type,
-    ...,
-    dependencies = dependencies,
-    quiet = quiet
+  do.call(
+    safe_install_packages,
+    c(list(
+        packages,
+        repos = repos,
+        type = type,
+        dependencies = dependencies,
```

Comment:
Do we need to translate `dependencies = TRUE` to `dependencies = NA` to avoid installing all suggested dependencies of dependencies?

Reply:
Based on an empirical test, this works fine, at least `install_github("r-lib/processx", dependencies = TRUE)` only installs `Suggests` for processx.